ui/src/shared/components/DashboardRoute.tsx

```diff
@@ -0,0 +1,57 @@
+import React, {PureComponent} from 'react'
+import {connect} from 'react-redux'
+import {withRouter, WithRouterProps} from 'react-router'
+import {setDashboard} from 'src/shared/actions/currentDashboard'
+import {AppState} from 'src/types'
+
+interface StateProps {
+  dashboard: string
+}
+
+interface DispatchProps {
+  updateDashboard: typeof setDashboard
+}
+
+type Props = StateProps & DispatchProps & WithRouterProps
+
+class DashboardRoute extends PureComponent<Props> {
+  check(props) {
+    const {dashboard, updateDashboard} = props
+    const dashboardID = props.params.dashboardID
+
+    if (dashboard !== dashboardID) {
+      updateDashboard(dashboardID)
+    }
+  }
+
+  componentDidMount() {
+    this.check(this.props)
+  }
+
+  componentWillUnmount() {
```

Comment:
You could do all of this with the `useEffect` hook in an `FC`

Reply:
i still don't know if this is the right way to do this.. so i'm trying to stay flexible. i think i want it as a decorator declared on the route, but.. here we are